components/automate-ui/src/app/services/projects-filter/projects-filter.reducer.spec.ts

```diff
@@ -0,0 +1,692 @@
+  ALL_PROJECTS_LABEL,
+  MULTIPLE_PROJECTS_LABEL
+} = ProjectConstants;
+
+const BADGE_GREY_STATE = false;
+const BADGE_BLUE_STATE = true;
+
+describe('projectsFilterReducer', () => {
+  const initialState: ProjectsFilterState = projectsFilterInitialState;
+
+  describe('undefined action', () => {
+    it('should return the default state', () => {
+      const action = { type: 'NOOP' } as any;
+      const result = projectsFilterReducer(undefined, action);
+
+      expect(result).toBe(initialState);
+    });
+  });
+
+  describe('LOAD_OPTIONS', () => {
+    const action = new LoadOptions();
+
+    it('sets status to loading', () => {
+      const { optionsLoadingStatus } = projectsFilterReducer(initialState, action);
+      expect(optionsLoadingStatus).toEqual(EntityStatus.loading);
+    });
+  });
+
+  describe('LOAD_OPTIONS_SUCCESS', () => {
+
+    it('sets status to success', () => {
+      const action = genAction(
+        genProject('p1')
+      );
+
+      const { optionsLoadingStatus } = projectsFilterReducer(initialState, action);
+
+      expect(optionsLoadingStatus).toEqual(EntityStatus.loadingSuccess);
+    });
+
+    describe('sorting options', () => {
+      it('sorts projects when unassigned is not included', () => {
+        const action = genAction(
+          genProject('d-proj'),
+          genProject('b-proj'),
+          genProject('c-proj'),
+          genProject('a-proj')
+        );
+
+        const { options } = projectsFilterReducer(initialState, action);
+
+        expect(map('label', options)).toEqual(['a-proj', 'b-proj', 'c-proj', 'd-proj']);
+      });
+
+      it('sorts projects except unassigned is put last', () => {
+        const action = genAction(
+          genProject('d-proj'),
+          genUnassignedProject(),
+          genProject('b-proj'),
+          genProject('zz-proj'),
+          genProject('a-proj')
+        );
+
+        const { options } = projectsFilterReducer(initialState, action);
+
+        expect(map('value', options))
+          .toEqual(['a-proj', 'b-proj', 'd-proj', 'zz-proj', UNASSIGNED_PROJECT_ID]);
+      });
+
+      it('sorts embedded numbers smartly', () => {
+        const action = genAction(
+          genProject('a-proj1'),
+          genProject('a-proj10'),
+          genProject('a-proj2')
+        );
+
+        const { options } = projectsFilterReducer(initialState, action);
+
+        expect(map('value', options))
+          .toEqual(['a-proj1', 'a-proj2', 'a-proj10']);
+      });
+
+      it('sorts first by case-independence then with lowercase before uppercase', () => {
+        const action = genAction(
+          genProject('a-proj'),
+          genProject('b-proj'),
+          genProject('A-proj')
+        );
+
+        const { options } = projectsFilterReducer(initialState, action);
+
+        expect(map('value', options))
+          .toEqual(['a-proj', 'A-proj', 'b-proj']);
+      });
+    });
+
+    describe('merging fetched and stored options', () => {
```

Comment:
New section of tests added to test merging the newly fetched options from the back-end with the browser-retained stored options.